### ISSUE TITLE
关于相关性图的计算

### ISSUE THREAD

**Issue by hcx163** [BBBBchan/CorrMatch#22]
作者你好，我看到在你的论文中描述的是相关性图由两个不同的弱增强图像经过编码器后分别得到w1和w2，然后将w1与w2相乘，但是我在您的代码中，发现实际上的操作是对同一个feature分别进行两个不同的卷积之后得到f1和f2，我想知道，这是否有冲突。
feature = F.interpolate(feature_in, (h_in, w_in), mode='bilinear', align_corners=True)
        f1 = rearrange(self.conv1(feature), 'n c h w -> n c (h w)')
        f2 = rearrange(self.conv2(feature), 'n c h w -> n c (h w)')
        out_temp = rearrange(out, 'n c h w -> n c (h w)')
        corr_map = torch.matmul(f1.transpose(1, 2), f2) / torch.sqrt(torch.tensor(f1.shape[1]).float())
        corr_map = F.softmax(corr_map, dim=-1)
